pkg/ee/metering/reconcile.go

```diff
@@ -83,3 +91,31 @@ func ReconcileMeteringResources(ctx context.Context, client ctrlruntimeclient.Cl
 
 	return nil
 }
+
+// cleanupMeteringResources removes active parts of the metering
+// components, in case the admin disables the feature.
+func cleanupMeteringResources(ctx context.Context, client ctrlruntimeclient.Client) error {
+	key := types.NamespacedName{Namespace: resources.KubermaticNamespace, Name: meteringToolName}
+	if err := cleanupResource(ctx, client, key, &appsv1.Deployment{}); err != nil {
+		return fmt.Errorf("failed to cleanup metering Deployment: %v", err)
+	}
+
+	key.Name = meteringCronJobWeeklyName
+	if err := cleanupResource(ctx, client, key, &batchv1beta1.CronJob{}); err != nil {
+		return fmt.Errorf("failed to cleanup metering CronJob: %v", err)
+	}
+
+	return nil
+}
+
+func cleanupResource(ctx context.Context, client ctrlruntimeclient.Client, key types.NamespacedName, obj ctrlruntimeclient.Object) error {
```

Comment:
So for the RBAC stuff I would leave it, but what about the secret and configurations? Such as the bucket endpoint, aws/minio credentials?

Reply:
I would also leave them.